### PR TITLE
BO-174 Add systemd support to fluentd

### PR DESCRIPTION
**Pull request by zazabe**

### CODE REVIEW

**Comment by zazabe** [comment]
mmmh, all of this is a bit messy, some commits are not visible in this review, but rather in https://upsource.cargomedia.ch/puppet-packages/review/PP-CR-26, even if this one is closed already...
Anyhow, all commits are included in https://upsource.cargomedia.ch/puppet-packages/branch/PR%201713

File: modules/fluentd/spec/source_journald/spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  describe file('/var/lib/fluentd/journald_pos_systemd') do
+    it { should be_directory }
+  end
+
+  describe command('grep foo /tmp/dump/*.log | tail -1') do
+    its(:exit_status) { should eq 0 }
+    its(:stdout) do
+      is_expected.to include_json(
```
Comment (ppp0) [inline modules/fluentd/spec/source_journald/spec.rb]: Coool

**Comment by ppp0** [comment]
spec test failure

**Comment by ppp0** [comment]
The idea is to selectively enable journal source for fluentd with hiera
```
  "systemd::config::journald::fluentd_output": true,
```

due to the fact that we don't know yet much about the log volume to be expected

**Pull request merged**